File: crates/domain/src/event_group.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

use crate::ID;

/// Group of calendar events
#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export)]
pub struct EventGroup {
    /// Unique ID
    pub id: ID,

    /// Calendar ID to which the group belongs
    pub calendar_id: ID,

    /// User ID
    pub user_id: ID,

    /// Account ID
    pub account_id: ID,

    /// Parent ID - this is an ID external to the system
    /// It allows to link groups of events together to an outside entity
    pub parent_id: Option<String>,

    /// External ID - this is an ID external to the system
    /// It allows to link a group of events to an outside entity
    pub external_id: Option<String>,
}

File: dotnet/src/Domain/EventGroup.cs
using Nittei.Domain.Shared;

namespace Nittei.Domain;

/// <summary>
/// Group of calendar events
/// </summary>
public class EventGroup : IEntity<Id>
{
  /// <summary>
  /// Unique ID
  /// </summary>
  public Id Id { get; set; }

  /// <summary>
  /// Calendar ID to which the group belongs
  /// </summary>
  public Id CalendarId { get; set; }

  /// <summary>
  /// User ID
  /// </summary>
  public Id UserId { get; set; }

  /// <summary>
  /// Account ID
  /// </summary>
  public Id AccountId { get; set; }

  /// <summary>
  /// Name of the event group
  /// </summary>
  public string Name { get; set; } = string.Empty;

  /// <summary>
  /// Description of the event group
  /// </summary>
  public string Description { get; set; } = string.Empty;

  /// <summary>
  /// Parent ID - this is an ID external to the system
  /// It allows to link groups of events together to an outside entity
  /// </summary>
  public string? ParentId { get; set; }

  /// <summary>
  /// External ID - this is an ID external to the system
  /// It allows to link a group of events to an outside entity
  /// </summary>
  public string? ExternalId { get; set; }

  public EventGroup()
  {
    Id = Id.NewId();
  }
}